lib/auth/register.go

```diff
@@ -152,6 +152,13 @@ func Register(params RegisterParams) (*proto.Certs, error) {
 		if err != nil {
 			return nil, trace.Wrap(err)
 		}
+		if !utils.IsEC2NodeID(params.ID.HostUUID) {
+			return nil, trace.BadParameter(
+				`host ID %q is not valid when using the EC2 join method, `+
+					`try removing the "host_uuid" file in your teleport data dir `+
+					`(e.g. /var/lib/teleport/host_uuid)`,
```

Comment:
Some minor nitpicks:
- I see other messages like this start the sentence with capitals, not sure what's the rule =p. https://github.com/gravitational/teleport/blob/63e17f8a0f9a21dd66666cbb1eef4f529684eca5/lib/service/service.go#L799-L802
- when this error happens, probably can skip `utils.GetEC2IdentityDocument()` above
- it will be great if we can print the actual UUID file path, but i would also understand if we don't want to "pollute" things here

Reply:
went ahead and capitalized it + moved the check to before we fetch the identity document. Don't want to plumb the actual data path down here at the moment.